tests/test_qaoa.py

```diff
@@ -764,3 +765,102 @@ def test_duplicates_remove_zeros(self):
         assert len(reduced_ops) == 1
         assert reduced_ops[0].name == "PauliZ"
         assert reduced_ops[0].wires.tolist() == [1]
+
+
+    def test_inner_net_flow_constraint_hamiltonian(self):
+        """Test if the _inner_net_flow_constraint_hamiltonian function returns the expected result on a manually-calculated
+        example of a 3-node complete digraph relative to the 0 node"""
+        g = nx.complete_graph(3).to_directed()
+        h = _inner_net_flow_constraint_hamiltonian(g, 0)
+
+        expected_ops = [
+            qml.Identity(0),
+            qml.PauliZ(0) @ qml.PauliZ(1),
+            qml.PauliZ(0) @ qml.PauliZ(2),
+            qml.PauliZ(0) @ qml.PauliZ(4),
+            qml.PauliZ(1) @ qml.PauliZ(2),
+            qml.PauliZ(1) @ qml.PauliZ(4),
+            qml.PauliZ(2) @ qml.PauliZ(4),
+        ]
+        expected_coeffs = [4, 2, -2, -2, -2, -2, 2]
+
+        assert expected_coeffs == h.coeffs
+        assert all([op.wires == exp.wires for op, exp in zip(h.ops, expected_ops)])
+
+
+    def test_inner_net_flow_constraint_hamiltonian_non_complete(self):
+        """Test if the _inner_net_flow_constraint_hamiltonian function returns the expected result on a manually-calculated
+        example of a 3-node complete digraph relative to the 0 node, with the (1, 0) edge removed"""
+        g = nx.complete_graph(3).to_directed()
+        g.remove_edge(1, 0)
+        h = _inner_net_flow_constraint_hamiltonian(g, 0)
+
+        expected_ops = [
+            qml.Identity(0),
+            qml.PauliZ(0),
+            qml.PauliZ(1),
+            qml.PauliZ(3),
+            qml.PauliZ(0) @ qml.PauliZ(1),
+            qml.PauliZ(0) @ qml.PauliZ(3),
+            qml.PauliZ(1) @ qml.PauliZ(3),
+        ]
+        expected_coeffs = [4, -2, -2, 2, 2, -2, -2]
+
+        assert expected_coeffs == h.coeffs
+        assert all([op.wires == exp.wires for op, exp in zip(h.ops, expected_ops)])
```

Comment:
I think the following line isn't working as expected:
```python
assert [h.ops[i] == expected_op for i, expected_op in enumerate(expected_ops)]
```
Actually:
```pycon
>>> print([h.ops[i] == expected_op for i, expected_op in enumerate(expected_ops)])
[False, False, False, False, False, False, False]
```
It is only passing the assertion because a non-empty list evaluates to `True`. Ideally we'd have
```python
assert all([h.ops[i] == expected_op for i, expected_op in enumerate(expected_ops)])
```
However, testing equality between operators doesn't work:
```pycon
>>> qml.PauliZ(0) == qml.PauliZ(0)
False
```
We therefore have to go back to comparing wires, as well as something like comparing `op.name` (which is sometimes a list if the op is a tensor).

Reply:
Addressed here ae5cc0d1d803263db75ecf0d37219f61ee530410 by comparing names of operators as strings:

```python
>>> for i, expected_op in enumerate(expected_ops): print(str(h.ops[i]) == str(expected_op))
True
True
True
True
True
True
True
```
 so we have the following assertion:

```python
for i, expected_op in enumerate(expected_ops): assert str(h.ops[i]) == str(expected_op)
```